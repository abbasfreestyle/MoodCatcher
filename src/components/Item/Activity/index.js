import React from 'react';
import PropTypes from 'prop-types';

import { View, Text, StyleSheet } from 'react-native';

import moment from 'moment';

import { moodPercentage } from 'utils/mood';

import Grid from 'components/Grid';
import Chip from 'components/Chip';
import Face from 'components/Face';

import theme from 'theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 10,
    borderRadius: 10,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: theme.secondary,
    minHeight: 50,
    shadowColor: theme.shadow,
    shadowOpacity: 0.5,
    shadowRadius: 3,
    shadowOffset: {
      height: 2,
      width: 0
    },
    elevation: 5
  },
  circle: {
    height: 60,
    width: 60,
    borderRadius: 30,
    backgroundColor: theme.surface
  },
  date: {
    color: theme.onSecondary
  },
  text: {
    color: theme.onSecondary,
    padding: 10
  }
});

const Activity = props => {
  const { item } = props;
  return (
    <View style={styles.container}>
      <Grid.Row flex={1}>
        <View padding={10} alignItems="center" width={75}>
          <View style={styles.circle}>
            <Face progress={moodPercentage(item.mood)} />
          </View>
        </View>
        <Grid.Column flex={1} justifyContent="center" padding={5}>
          <Grid.Row padding={5}>
            <Text style={styles.date}>
              {moment(item.date).format('MMMM DD - h:mm A')}
            </Text>
          </Grid.Row>
          <Grid.Row flexWrap="wrap" padding={5}>
            {item.feelings.items.map(feeling => (
              <Chip
                key={feeling.id}
                text={feeling.name}
                backgroundColor={theme.surface}
              />
            ))}
          </Grid.Row>
          <Text style={styles.text}>{item.comment}</Text>
        </Grid.Column>
      </Grid.Row>
    </View>
  );
};

Activity.propTypes = {
  item: PropTypes.object.isRequired
};

export default Activity;
