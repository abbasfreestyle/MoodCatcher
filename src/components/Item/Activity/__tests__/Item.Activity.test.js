import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Activity from '..';

const requiredProps = {
  item: {
    mood: 6,
    date: new Date(2019, 2, 26).valueOf(),
    feelings: {
      items: [{ id: 1, name: 'Relaxed' }, { id: 2, name: 'Happy' }]
    },
    comment: 'Test comment'
  }
};

const wrapper = shallow(<Activity {...requiredProps} />);

describe('Item > Activity.', () => {
  describe('Snapshots.', () => {
    test('should match', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
