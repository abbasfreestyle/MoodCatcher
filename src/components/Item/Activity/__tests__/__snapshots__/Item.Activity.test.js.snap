// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item > Activity. Snapshots. should match 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F2FCFB",
      "borderRadius": 10,
      "elevation": 5,
      "flex": 1,
      "justifyContent": "center",
      "margin": 10,
      "minHeight": 50,
      "shadowColor": "#44BBA4",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 3,
    }
  }
>
  <Row
    flex={1}
  >
    <View
      alignItems="center"
      padding={10}
      width={75}
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 30,
            "height": 60,
            "width": 60,
          }
        }
      >
        <Face
          progress={0.8333333333333334}
          size={75}
        />
      </View>
    </View>
    <Column
      flex={1}
      justifyContent="center"
      padding={5}
    >
      <Row
        padding={5}
      >
        <Text
          style={
            Object {
              "color": "#393E41",
            }
          }
        >
          March 26 - 12:00 AM
        </Text>
      </Row>
      <Row
        flexWrap="wrap"
        padding={5}
      >
        <Chip
          backgroundColor="#FFFFFF"
          key="1"
          text="Relaxed"
        />
        <Chip
          backgroundColor="#FFFFFF"
          key="2"
          text="Happy"
        />
      </Row>
      <Text
        style={
          Object {
            "color": "#393E41",
            "padding": 10,
          }
        }
      >
        Test comment
      </Text>
    </Column>
  </Row>
</View>
`;
